pkg/util/tolerations/tolerations.go

```diff
@@ -0,0 +1,126 @@
+/*
+Copyright 2016 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package tolerations
+
+import (
+	"k8s.io/kubernetes/pkg/api"
+)
+
+type key struct {
+	tolerationKey string
+	effect        api.TaintEffect
+}
+
+// VerifyAgainstWhitelist checks if the provided tolerations
+// satisfy the provided whitelist and returns true, otherwise returns false
+func VerifyAgainstWhitelist(tolerations []api.Toleration, nsWhiteList []api.Toleration) bool {
+	if len(nsWhiteList) == 0 {
+		return true
+	}
+
+	t := ConvertTolerationToAMap(tolerations)
+	w := ConvertTolerationToAMap(nsWhiteList)
+
+	for k1, v1 := range t {
+		if v2, ok := w[k1]; !ok || !AreEqual(v1, v2) {
+			return false
+		}
+	}
+	return true
+}
+
+// IsConflict returns true if the key of two tolerations match
+// but one or more other fields differ, otherwise returns false
+func IsConflict(first []api.Toleration, second []api.Toleration) bool {
+	firstMap := ConvertTolerationToAMap(first)
+	secondMap := ConvertTolerationToAMap(second)
+
+	for k1, v1 := range firstMap {
+		if v2, ok := secondMap[k1]; ok && !AreEqual(v1, v2) {
+			return true
+		}
+	}
+	return false
+}
+
+// MergeTolerations merges two sets of tolerations into one
+// it does not check for conflicts
```

Comment:
yes. But in this plugin, its not a concern as we check for conflicts before calling merge.